src/Sylius/Bundle/PaymentBundle/Checker/PaymentRequestFinalTransitionChecker.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\PaymentBundle\Checker;
+
+use Sylius\Abstraction\StateMachine\StateMachineInterface;
+use Sylius\Component\Payment\Model\PaymentRequestInterface;
+use Sylius\Component\Payment\PaymentRequestTransitions;
+
+final class PaymentRequestFinalTransitionChecker implements PaymentRequestFinalTransitionCheckerInterface
```

Comment:
```suggestion
final class CompletedPaymentRequestChecker implements CompletedPaymentRequestCheckerInterface
```
A bit more ambiguity and the method would be `isCompleted`

Reply:
@NoResponseMate From the state machine perspective it's a final state, "completed" rely to the actual payment request state but another state can be set by developers here.

Reply:
```suggestion
final class PaymentRequestFinalStateChecker implements PaymentRequestFinalStateCheckerInterface
```

What about replacing "Transition" to "State" ?

Reply:
Ah, I forgot that there is an actual `completed` state. I meant it in a way that there is no more processing that can be done on a payment request, the fact that this implementation relies on a state machine is just an implementation detail. Maybe something like `FinalizedPaymentRequestChecker`? This suggestion applies more to the interface though.

Reply:
I like it ;)

Let's rename both of them then

```suggestion
final class FinalizedPaymentRequestChecker implements FinalizedPaymentRequestCheckerInterface
```